cli/src/commands/init/index.ts

```diff
@@ -157,14 +171,13 @@ export default class Init extends BaseCommand<typeof Init> {
       await this.installPackage(packageManager, '@xata.io/client');
     }
 
-    const branch = this.getCurrentBranchName();
     if (schema) {
       await this.deploySchema(workspace, region, database, branch, schema);
     }
 
     // Run pull to retrieve remote migrations, remove any local migrations, and generate code
     await Pull.run([branch, '-f', '--skip-code-generation']);
-    await Codegen.runIfConfigured(this.projectConfig);
+    await Codegen.runIfConfigured(this.projectConfig, [`--branch=${branch}`]);
```

Comment:
I left out the `--inject-branch` flag, we didn't write `main` into the output here so I didn't write the selected branch in their either.